### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ciricode**
- check which session is running to give the correct message

### CODE REVIEW

**Review by rivanfebrian123 — CHANGES_REQUESTED** [review]
Don't remove those OS detection. Those warnings are for specific distros, not all distro with the same DE

**Comment by ciricode** [comment]
> Don't remove those OS detection. Those warnings are for specific distros, not all distro with the same DE

I don't think so, a distro can use any DE. The fact that I am using ubuntu and I install xfce4, and now I have 2 DE on 1 distro

**Comment by rivanfebrian123** [comment]
Yeah, I do appreciate your work tho, but, you know, I used to be like you too..., using more than one DE on an OS. And you know what? They break my WM theming, especially on Inkscape on GNOME

**Comment by ciricode** [comment]
> Yeah, I do appreciate your work tho, but, you know, I used to be like you too..., using more than one DE on an OS. And you know what? They break my WM theming, especially on Inkscape on GNOME

As far as I understand the code is just to check which DE is used to display the appropriate message, I don't really understand what role Distro plays in the code. For example Arch Linux we can't determine which DE the user is using.

**Comment by ging-dev** [comment]
> As far as I understand the code is just to check which DE is used to display the appropriate message, I don't really understand what role Distro plays in the code. For example Arch Linux we can't determine which DE the user is using.
What if DE is determined based on an environment variable like $XDG_CURRENT_DESKTOP or $DESKTOP_SESSION?

**Comment by ging-dev** [comment]
> Yeah, I do appreciate your work tho, but, you know, I used to be like you too..., using more than one DE on an OS. And you know what? They break my WM theming, especially on Inkscape on GNOME

There are many distros using xfce4 like `manjaro`, `xubuntu`, `linux mint`, `linux lite`, `gentoo`,... etc. I'm not sure if checking via distro will work.

**Comment by ciricode** [comment]
> What if DE is determined based on an environment variable like $XDG_CURRENT_DESKTOP or $DESKTOP_SESSION?

Will environment variables work on all distro? I guess not, need verification.

**Comment by rivanfebrian123** [comment]
Well, the problems only happen on some distro, not all distro. That's why we need `is_my_distro` here

**Comment by ging-dev** [comment]
> Chà, các vấn đề chỉ xuất hiện trên một số bản phân phối, không phải tất cả các bản phân phối. Đó là lý do tại sao chúng ta cần `is_my_distro`ở đây



> Well, the problems only happen on some distro, not all distro. That's why we need `is_my_distro` here

No, panel opacity display occurs mostly on xfce versions 4.14 and sometimes still on 4.16, i don't think only in some distro.

**Comment by rivanfebrian123** [comment]
When I tested it (XFCE theme) on Fedora and Ubuntu, it worked just fine, unlike Arch or Void that require logout (or kill the `xfce-panel`) to solve this

**Comment by ging-dev** [comment]
> When I tested it (XFCE theme) on Fedora and Ubuntu, it worked just fine, unlike Arch or Void that require logout (or kill the `xfce-panel`) to solve this

I think it depends on the version of Xfce4 you are using. I use xfce 4.12 (xubuntu 18.04) then upgrade to xfce 4.14 (xubuntu 20.04), the panel display error sometimes still occurs. It is better improved on xfce 4.16. I think you compare xfce4 on 2 different versions.

**Comment by rivanfebrian123** [comment]
Well, you can make some version checks then :)

**Comment by ging-dev** [comment]
> Well, you can make some version checks then :)

well, i think that warning should appear on every version of xfce4, i can reproduce this error on latest version of xfce4: https://streamable.com/zr2ab7

File: install.sh [hunk 1]
```diff
@@ -150,11 +150,11 @@ else
 
   # rm -rf "${THEME_SRC_DIR}/sass/_gtk-base-temp.scss"
 
-  if (is_my_distro "arch" || is_my_distro "void") && has_command xfce4-session; then
-    msg="XFCE: you may need to logout after changing your theme to fix your panel opacity."
-  elif (is_my_distro "solus") && has_command gnome-shell; then
+  if (is_running "xfce4-session"); then
+    msg="XFCE: you may need to run 'xfce4-panel -r' after changing your theme to fix your panel opacity."
+  elif (is_running "gnome-session") && [[ "${GNOME_VERSION}" == "new" ]]; then
```
Comment (ging-dev) [inline install.sh:155]:
```patch
- elif (is_my_distro "solus") && has_command gnome-shell; then
+ elif (is_running "gnome-session") && [[ "${GNOME_VERSION}" == "new" ]]; then
```
Please explain this line

Comment (ciricode) [reply install.sh:155]: You can easily do a google search for information Solus is starting to use GNOME 40

Comment (ging-dev) [reply install.sh:155]: I think solus has its own creations different from Gnome 40, you should revert it

**Pull request merged**